packages/compiler/core/checker.ts

```diff
@@ -677,7 +715,19 @@ export function createChecker(program: Program): Checker {
     args: Type[]
   ): Type {
     const symbolLinks = getSymbolLinks(templateNode.symbol);
-    const cached = symbolLinks.instantiations!.get(args) as ModelType;
+    if (symbolLinks.instantiations === undefined) {
+      const type = getTypeForNode(templateNode);
+      if (isErrorType(type)) {
+        return errorType;
+      } else {
+        throw new Error(
```

Comment:
What is the impact of throwing an error here?  Does it end checker execution and allow the language server to recover?  If not, do we want the language server itself to throw an exception / crash under these circumstances (i.e. this is totally unrecoverable)?

Reply:
So right now it was crashing with accessing undefined property. I really don't know when this scenario can happen apart from the isErrorType which I covered but this is to have a better error message in case it ever does that another undefined error.
@nguerrera fixed the language server to recover on crashes like that.

Reply:
This is OK, but I suggest compilerAssert(false,... for cases where you believe something can't happen but want a good message if it ever does. This makes it easier to scrub and make sure we aren't using exceptions for diagnostics.